### DISCUSSION TITLE
Add support for playing Revohloo music videos

### DISCUSSION THREAD

**Discussion by payini** [deso-protocol/dips#85] [Medium CIPs]
I would like to add the ability to post Revohloo music videos and content from https://revohloo.com, in the same way YouTube, Vimeo, SoundCloud, and other platforms are supported.

Currently YouTube, Vimeo, TikTok, Giphy, Spotify and SoundCloud are supported, but not Revohloo, an interactive music video streaming platform. This change would allow us to post music videos, promote artists in BitClout, and potentially attract around 5K+ artists to join BitClout.

I already wrote about 90% of the changes needed for this support on my local development environment, and the videos play successfully already while posting the video. I only need to find out the changes needed for the videos to play in the feed (currently they are not playing nor showing up once posted,) as well as to add some unit tests to be consistent with the exiting code.

Full disclosure: I am the Co-Founder and CTO of https://revohloo.com.

**Comment by TracyDeBarros** [comment]
Good Morning 🌄 BitClout Development!

First thing is, Team Revohloo is UBER Excited to be apart this Amazing Experiment! We believe 100% in the Mission laid out By Diamond Hands & Team.

We've had our music video platform for Several Years, but we had NO SOCIAL LAYER. So when we Discovered BitClout , we were and Remain Blow Away.

It's like BitClout was built for Revohloo.

Adding Player Support for Revohloo Music Videos and Non Video Content will definitely Add Value across the NETWORK.

We currently use VideoJS.com Player the world's most popular open source HTML5 player framework.

We're planning Live Events broadcast Free on the NETWORK. Not having to rely on YouTube to post High Quality Video Content will be a Major Benefit to the Entire Network.

Posting video content on Revohloo has always been and will remain FREE. Creators will earn their 💎 Rewards! And we won't have broken YouTube redirect links to view BitClout Creator Content on YouTube!
 
We look forward to moving our Project forward and assisting with any other projects which require Video.

Infact, we've reached out to Maz & @BrixBoston100 about creating a series of short Explainer Videos to educate New Creators on how to properly leverage the Platform.

We plan on bringing Thousands of Independent recording artists, so we feel it is our Responsibility to educate them. Their success is our SUCCESS. The responsibility is NOT just Diamond Hands and the BitClout Team.

TRUE DECENTRALIZATION MEANS THE RESPONSIBILITY BELONGS TO ALL!!!

Thank You and Let's Grow 🌱🌱🌱 BitClout!

Tracy DeBarros
Founder/CEO/Hustler
Revohloo.live
Powered by💲CLOUT

Call me Direct 24/7 800.587.1702

**Comment by omicreativedev** [comment]
I would love to see Revohloo and Twitch supported across all nodes. If Revohloo is the only node to support them then it will be exclusive. Either way I think it's a win for Revohloo. I'm excited about this project and look forward to re-enaging with the active userbase and bringing the video tech to the music, art, and film space. It's truly one of a kind.

**Comment by bloated-devish** [comment]
> I already wrote about 90% of the changes needed for this support on my local development environment, and the videos play successfully already while posting the video. I only need to find out the changes needed for the videos to play in the feed (currently they are not playing nor showing up once posted,) as well as to add some unit tests to be consistent with the exiting code.

Did you check the  recent PR for adding additional data sources? You could reuse them. Did you check the JS console logs, may be cross site reference are preventing it from being rendered? Overall, I think the effort to do this should not be high. At some point, I think we should probably have 50-100 different sources of embedding data, as long as we can convince these data sources don't end up injecting something and creating mess (like downloading a malware for example).

**Reply by payini** [reply]
Thanks for replying @bloated-devish 

I haven't seen that data sources PR. I just looked for it in frontend, backend and core repos, but I can't find it. Can you share that PR? Agree is not a high effort, seems small to me, @maebeam told me to log it as medium.

I did check the JS console logs, below is the error I'm getting, if you could provide any help/direction I'll appreciate it.

This is a sample post where the video plays successfully before posting:

![image](https://user-images.githubusercontent.com/2638642/126121695-2b552514-3f9b-49fb-b59a-4a6a9a37998f.png)

And this is the way it shows up in the feed:

![image](https://user-images.githubusercontent.com/2638642/126122190-c58d7a6b-9ff3-4ea2-a088-aaf92611d2fa.png)

The error I'm getting in the JS console logs, when trying to view the video from the feed (picture above) is this one, but I'm not sure if is related:

![image](https://user-images.githubusercontent.com/2638642/126122330-0e0c91fb-be5e-496b-99ca-7ea4f9e1a674.png)

Not sure if I need to add the revohloo.com domain somewhere else, I currently just added it to the Caddyfile, but not sure what I'm missing.

Any feedback would be really appreciated.

**Reply by payini** [reply]
Ok I got the code all figured it out and completed. Ready for the next step.

**Reply by payini** [reply]
Revohloo videos play on the feed as well now:

![image](https://user-images.githubusercontent.com/2638642/126888425-9e9461f0-f112-4792-9764-f1a030fa9b02.png)

**Reply by lazynina** [reply]
I think adding support for revohloo would be a good. You may proceed. 

Eventually we'll need some way to convey to users which sources are supported.

**Reply by payini** [reply]
Awesome, I will be submitting a PR very soon. If I understand the comment correctly, I guess the input box placeholder is a way to convey to the users what sources are supported.

![image](https://user-images.githubusercontent.com/2638642/126947938-00c9621b-f4fa-4146-85bc-9085ca51378f.png)

**Comment by payini** [comment]
@lazynina I went ahead and submitted a PR last night: https://github.com/bitclout/frontend/pull/322

**Comment by diamondhands0** [comment]
Hey everyone, I'm very sorry to slow things down here but we are currently concerned that integrating more third-party video services will be a security risk for users. In particular, any domain that gets whitelisted in the reference node implementation gives that domain owner the ability to track BitClout users, and potentially take them to malicious landing pages. As such, we need to be very thoughtful in how we go about whitelisting third-party providers. I think Revohloo is an awesome platform, but we will need more time to think about the policy around this. I'm sorry this is not a more concrete answer, but I didn't want to overpromise or compromise on security here. Hopefully this makes sense.

**Reply by payini** [reply]
Hello @diamondhands0,

Not a slow down at all, thanks for replying and expressing your concerns regarding security. They indeed make sense.

Just to be clear about the proposed changes to integrate Revohloo videos into the reference node implementation, I am the co-founder of Revohloo along with founder @Revohloo, and as such, we are willing to comply with any security policies you guys come up with and adhere to any security standards you may request from us.

We are proud to say that over the past several years Revohloo has securely collected and stored personal data from over 5,000 artists and have not suffered any breach of security.

With that said, we will appreciate if you please let us know, at your earliest convenience, when you guys have time to define such policy, as we would be happy to comply to it, and hopefully become a first-class citizen of BitClout.com.

Best regards.
